src/Merger/SchemaMerger.php

```diff
@@ -156,13 +156,21 @@ public function exportSchema(
 
         $prefix = '';
 
-        if (true === $prefixWithNamespace && false === $rootSchemaTemplate->isPrimitive()) {
+        if (
+            true === $prefixWithNamespace
+            && false === $rootSchemaTemplate->isPrimitive()
+            && is_array($rootSchemaDefinition)
```

Comment:
```suggestion
```
I think it would be probably nicer to enhance `SchemaTemplate::isPrimitive` to work with both string and json, wdyt?
I assume this new check is trying to cover this, correct?

Reply:
Yeah, that's right. I agree that we should enhance `SchemaTemplate::isPrimitive` instead to allow check for both string and json.